node/pkg/node/publicrpcRunnable.go

```diff
@@ -15,21 +16,28 @@ import (
 	"google.golang.org/grpc"
 )
 
-func publicrpcTcpServiceRunnable(logger *zap.Logger, listenAddr string, publicRpcLogDetail common.GrpcLogDetail, db *db.Database, gst *common.GuardianSetState, gov *governor.ChainGovernor) (supervisor.Runnable, error) {
-	l, err := net.Listen("tcp", listenAddr)
+func publicrpcTcpServiceRunnable(logger *zap.Logger, listenAddr string, publicRpcLogDetail common.GrpcLogDetail, db *db.Database, gst *common.GuardianSetState, gov *governor.ChainGovernor) supervisor.Runnable {
+	return func(ctx context.Context) error {
+		l, err := net.Listen("tcp", listenAddr)
```

Comment:
What happened to the deferred `l.close` you had added?

Reply:
that's actually already in `supervisor.GRPCServer(grpcServer, l, false)`.